src/js/tabs/osd.js

```diff
@@ -285,6 +222,220 @@ FONT.replaceLogoFromImage = function(img) {
+                            satisfied = constraint.test(img);
+                        if (satisfied) {
+                            LogoManager.showConstraintSatisfied(constraint);
+                        } else {
+                            LogoManager.showConstraintNotSatisfied(constraint);                        
+                            reject();
+                            return;
+                        }
+                    }
+                    resolve();
+                });
+            },
+            dialogOptions = { 
+                type: 'openFile', 
+                accepts: LogoManager.acceptFileTypes 
+            };
+
+        chrome.fileSystem.chooseEntry(dialogOptions, function(fileEntry) {
+            if (chrome.runtime.lastError) {
+                console.error(chrome.runtime.lastError.message);
+                return;
+            }
+            // load and validate selected image
+            var img = new Image();
+            img.onload = function() {
+                validateImage(img).then(function() {
+                    resolve(img);
+                }).catch(function(error) {
+                    reject(error);
+                });
+            };
+            img.onerror = function(error) {
+                reject(error);
+            };
+            fileEntry.file(function(file) {
+                img.src = "file://" + file.path;
+            });
+        });
+    });
+};
+
+/**
+ * Draw the logo using the loaded font data.
+ */
+LogoManager.drawPreview = function() {
+    var $el = LogoManager.elements.$preview;
+    $el.empty();
+    for (var i = SYM.LOGO, I = FONT.constants.MAX_CHAR_COUNT; i < I; i++) {
+        var url = FONT.data.character_image_urls[i];
+        $el.append('<img src="' + url + '" title="0x' + i.toString(16) + '"></img>');
+    }
+};
+
+/**
+ * Initialize Logo Manager UI.
+ */
+LogoManager.init = function() {
+    // cache DOM elements
+    Object.keys(LogoManager.elements).forEach(function(key) {
+        LogoManager.elements[key] = $(LogoManager.elements[key]);
+    });
+    Object.keys(LogoManager.constraints).forEach(function(key) {
+        LogoManager.constraints[key].$el = $(LogoManager.constraints[key].$el);
+    });
+    // resize logo preview area to match tile size
+    var logoWidthPx = LogoManager.constraints.imageSize.expectedWidth,
+        logoHeightPx = LogoManager.constraints.imageSize.expectedHeight;
+    LogoManager.elements.$preview
+        .width(logoWidthPx)
+        .height(logoHeightPx);
+    // inject logo size variables for dynamic translation strings
+    var takeFirst = obj => {
+        if (obj.hasOwnProperty("length") && 0 < obj.length) {
+            return obj[0];
+        } else {
+            return obj;
+        }
+    };
+    var lang = takeFirst(i18next.options.fallbackLng),
+        ns = takeFirst(i18next.options.defaultNS);
+    i18next.addResourceBundle(lang, ns, {
```

Comment:
I'm not too sure about this. We are using and encapsulating the localization framework under the localization.js file. Use the i18next library directly does not seem a good idea.
What are you trying to do with this?